x-pack/plugins/security_solution/public/common/components/event_details/insights/insights.tsx

```diff
@@ -45,14 +49,14 @@ export const Insights = React.memo<Props>(
       { category: 'kibana', field: 'kibana.alert.rule.parameters.index' },
       data
     );
-    const hasProcessEntityInfo =
-      isRelatedAlertsByProcessAncestryEnabled && processEntityField && processEntityField.values;
+    const hasProcessEntityInfo = processEntityField && processEntityField.values;
 
     const processSessionField = find(
       { category: 'process', field: 'process.entry_leader.entity_id' },
       data
     );
-    const hasProcessSessionInfo = processSessionField && processSessionField.values;
+    const hasProcessSessionInfo =
```

Comment:
if the length of `processSessionField.values` in the following check is zero:

```ts
    const hasProcessSessionInfo =
      isRelatedAlertsByProcessAncestryEnabled && processSessionField && processSessionField.values;
```

should `hasProcessSessionInfo` still be true?

(The same question may also apply to the other places in this file where `.values` is tested for truthiness.)

Reply:
Similar to your comment above, moved this to a type guard. Nice catch :) https://github.com/elastic/kibana/pull/138419/commits/24131f7be52465b2863577638e32d14dfa21443b